src/middleware.ts

```diff
@@ -146,10 +147,11 @@ async function sendTransform(
   req: IncomingMessage,
   res: ServerResponse,
   file: string,
-  transform: TransformFunc
+  transform: TransformOptionObject
 ): Promise<boolean> {
-  const content = await fs.readFile(file, 'utf8')
-  const transformedContent = transform(content, file)
+  const content = (await fs.readFile(file, transform.encoding)) as string &
+    Buffer
```

Comment:
This is sketchy, but the `fs.readFile` signature is wrong, it defines that `binary` can return a `string` type, so we need to refine it here.